### ISSUE TITLE
Link multiple players?

### ISSUE THREAD

**Issue by Gouster4** [spectran/CDJ-100S-STM32F7-Display#3]
Hello. Is there some way to link multiple players together to get:

-Shared content (tracks from sdcard insertednin 1st player will appear on 2nd player and vice versa)
-BPM and beat algin Synchronization

I thought about some 3.5mm jack connected between consoles (ideally 2 on each console to be able to connect 4 or more together).
I think that TTL (RX, TX, and Gnd) on 3.5mm jack can do the job. But it must be done also on software.

**Comment by spectran** [comment]
Yes, it is possible to make the link between players and sync enable. But I think, that UART is too weak interface to transmit the sound data from master player to slave player. Your idea with 3.5 mm jack with UART signal will be work only for BPM synchronization. I have an idea to look to USB connection between STM32F7-Discovery boards - they have two free USB micro connectors. And I need to modify code to transmit sound and and other data via USB.

**Comment by Gouster4** [comment]
Would be awesome. Maybe ethernet would be even better solution, as it can be used to link multiple devices using ethernet switch. Just like modern Pioneer players do, even with PC.

Library and bpm sync over ethernet between STMs would be cool, but reverse engineering of Pioneer Pro DJ link protocol would be even better. But i bet much much much harder. But this way you can combine them with newer pioneer players.
